cpp/oneapi/dal/table/backend/csr_table_impl.hpp

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/table/backend/csr_kernels.hpp"
+#include "oneapi/dal/table/detail/sparse_access_iface.hpp"
+
+namespace oneapi::dal::backend {
+
+class csr_table_impl : public detail::table_template<detail::csr_table_iface, csr_table_impl> {
+public:
+    csr_table_impl()
+            : col_count_(0),
+              row_count_(0),
+              layout_(data_layout::row_major),
+              csr_indexing_(detail::csr_indexing::one_based) {}
```

Comment:
Don't we need to create `detail::csr_indexing::default` value?

Reply:
I suppose we don't have to create it